dpgen/dispatcher/Dispatcher.py

```diff
@@ -406,3 +407,81 @@ def mdata_arginfo() -> List[Argument]:
     return [
         command_arginfo, machine_arginfo, resources_arginfo,
     ]
+
+
+def make_submission_compat(
+        machine: dict,
+        resources: dict,
+        commands: List[str],
+        work_dir: str,
```

Comment:
`work_dir` is the older version. In `dpdispatcher` version, this is `work_path`. Maybe we should let `work_path` be default?